imscjs-demo/js/index.js

```diff
@@ -1,59 +1,48 @@
-
-function initTTML() {
-  var renderDiv = document.getElementById("render-div");
-  var myVideo = document.getElementById("imscVideo");
-  //Track object generated by the first <track> child element
-  var myTrack = myVideo.textTracks[0];
-  //Src is only available from the <track> DOM element, not the obj
-  var ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
-  //The track is disabled by default, 'hidden' does not display VTT Text
-  //but fires the events
+async function initTTML() {
+  const renderDiv = document.getElementById("render-div");
+  const myVideo = document.getElementById("imscVideo");
+  // Track object generated by the first <track> child element
+  const myTrack = myVideo.textTracks[0];
+  // The source is only available from the <track> DOM element, not the object
+  const ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
+  // The track is disabled by default, 'hidden' does not display VTT Text
+  // but fires the events
   myTrack.mode = "hidden";
 
-  var client = new XMLHttpRequest();
-
-  client.open('GET', ttmlUrl);
-  client.onreadystatechange = function () {
-    initTrack(client.responseText);
-  }
-  client.send();
-
+  const response = await fetch(ttmlUrl);
+  initTrack(await response.text());
 
   function initTrack(text) {
-    var imscDoc = imsc.fromXML(text);
-    var timeEvents = imscDoc.getMediaTimeEvents();
-    //create cue per timed event and render isd
-    for (var i = 0; i < timeEvents.length; i++) {
-      //Edge/IE implement "Generic Cue", other browsers VTTCue
-      var Cue = window.VTTCue || window.TextTrackCue;
+    const imscDoc = imsc.fromXML(text);
+    const timeEvents = imscDoc.getMediaTimeEvents();
+    // Create one cue per timed event and render isd
+    for (let i = 0; i < timeEvents.length; i++) {
+      let myCue;
       if (i < timeEvents.length - 1) {
-        //We have to provide empty string as VTTText
-        var myCue = new Cue(timeEvents[i], timeEvents[i + 1], "");
+        // We have to provide an empty string as the VTTText
+        myCue = new VTTCue(timeEvents[i], timeEvents[i + 1], "");
```

Comment:
I think the standard is `TextTrackCue`, doesn't it?

Reply:
I know basically nothing about this API, but it looks to me like you can't construct a `TextTrackCue`, it doesn't have a constructor. From https://developer.mozilla.org/en-US/docs/web/api/texttrackcue:

> TextTrackCue is an abstract class which is used as the basis for the various derived cue types, such as [VTTCue](https://developer.mozilla.org/en-US/docs/Web/API/VTTCue); you will instead work with those derived types.

Reply:
I have tested this demo in Firefox and Chrome with VTTCue and it works, and with TextTrackCue, and it doesn't.

Reply:
👍